pkg/neotest/basic.go

```diff
@@ -401,6 +401,10 @@ func TestInvoke(bc *core.Blockchain, tx *transaction.Transaction) (*vm.VM, error
 	ttx := *tx
 	ic, _ := bc.GetTestVM(trigger.Application, &ttx, b)
 
+	if isCoverageEnabled() {
+		ic.VM.SetOnExecHook(coverageHook())
```

Comment:
Speaking of parallel execution: the whole `coverage*` functions operate on global variables. 
It seems we can create some `CoverageContext` here, provide its method to the VM and then process the results in `defer`. This way, our only conflict between different goroutines will be write to file.

Reply:
We can add mutex that locks when the file is being written, or vm executes next instruction. The major concern is speed, but currently the bottleneck is actually writing the file (during each test cleanup).